### PR TITLE
Add evaluation script.

### PR DESCRIPTION
**Pull request by hgaiser**
This PR adds an evaluation script for arbitrary datasets (meaning currently it adds support for the CSV generator dataset).

### CODE REVIEW

File: keras_maskrcnn/bin/evaluate.py [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+        os.makedirs(args.save_path)
+
+    # create the generator
+    generator = create_generator(args)
+
+    # load the model
+    print('Loading model, this may take a second...')
+    model = models.load_model(args.model, backbone_name=args.backbone)
+
+    # print model summary
```
Comment (vcarpani) [inline keras_maskrcnn/bin/evaluate.py]: Maybe it might be useful to add a verbose option that also print this

File: keras_maskrcnn/utils/eval.py [hunk 2]
```diff
@@ -0,0 +1,258 @@
...
+
+                if max_overlap >= iou_threshold and assigned_annotation not in detected_annotations:
+                    false_positives = np.append(false_positives, 0)
+                    true_positives  = np.append(true_positives, 1)
+                    detected_annotations.append(assigned_annotation)
+                else:
+                    false_positives = np.append(false_positives, 1)
+                    true_positives  = np.append(true_positives, 0)
+
+        # no annotations -> AP for this class is 0 (is this correct?)
```
Comment (vcarpani) [inline keras_maskrcnn/utils/eval.py]: I am not sure about this, when you do the average of the AP you divide for the total number of classes, shouldn't it be 1?

Comment (hgaiser) [reply keras_maskrcnn/utils/eval.py]: Neither would be correct actually. 0 would indicate something is wrong, 1 would indicate it is perfect, but it is neither..

File: keras_maskrcnn/bin/evaluate.py (hunk 1 above)
Comment (hgaiser) [reply keras_maskrcnn/bin/evaluate.py]: Hmm I have doubts about it. When do you think it'd be useful?

Comment (vcarpani) [reply keras_maskrcnn/bin/evaluate.py]: When debbugging changes to the network (for example what happened to @enricoliscio when workign on the regression), but it's a very uncommon case

File: keras_maskrcnn/utils/eval.py (hunk 2 above)
Comment (vcarpani) [reply keras_maskrcnn/utils/eval.py]: Can we check how other evaluation scripts do?

File: keras_maskrcnn/bin/evaluate.py (hunk 1 above)
Comment (hgaiser) [reply keras_maskrcnn/bin/evaluate.py]: Hmm, I propose to add that when it comes up.

File: keras_maskrcnn/utils/eval.py (hunk 2 above)
Comment (hgaiser) [reply keras_maskrcnn/utils/eval.py]: This method was used in py-faster-rcnn, that's the only reference I have :).

**Pull request merged**